src/sentry/notifications/helpers.py

```diff
@@ -834,9 +832,9 @@ def get_setting_options_for_users(
     if not additional_filters:
         additional_filters = Q()
 
-    users = user_service.get_many(filter={"user_ids": user_ids})
-
-    query = get_query(recipients=users, project=project, organization=organization)
+    users = User.objects.filter(id__in=user_ids)
+    projects = [project] if project else []
```

Comment:
Having access to both `User` and `Project` will not be possible in siloed application code. I think you might need to continue using `user_service` in notification logic that could run in region logic.